test/schema.test.js

```diff
@@ -2792,4 +2792,16 @@ describe('schema', function() {
       });
     }, /Cannot use schema-level projections.*subdocument_mapping.not_selected/);
   });
+
+  it('allows a lean option on schemas so that all documents are lean when running a query (gh-10090)', async function() {
+    const testSchema = new mongoose.Schema({
+      name: String
+    }, { lean: true });
+    const Test = db.model('gh10090', testSchema);
+    await Test.create({
+      name: 'I am a lean doc, fast and small'
+    });
+    const entry = await Test.findOne();
+    assert.equal(entry instanceof mongoose.Document, false);
```

Comment:
Add an extra check in this test that verifies you can still overwrite `lean()`.

So `const doc = await Test.findOne().lean(false); assert.ok(entry instanceof mongoose.Document)`

Also, no need to do  `assert.equal(entry instanceof mongoose.Document, false);`. Just `assert.ok(!entry instanceof mongoose.Document);`